lib/bot.js

```diff
@@ -98,6 +100,24 @@ class Bot {
       this.sendToDiscord(author, to, `*${text}*`);
     });
 
+    this.ircClient.on('join', (channel, nick) => {
+      if (this.disableJoinPartMsgs) return;
+      if (nick === this.nickname && !this.announceSelfJoin) return;
```

Comment:
In that case, wouldn't it make more sense to save up messages and send them when the IRC connection comes online? Maybe not if it's more than say 10 messages, but in your case I guess it could be a useful feature. Might not be worth the extra code though.